bluesky_env/plans/top_view.py

```diff
@@ -0,0 +1,133 @@
+import numpy as np
+from scipy.interpolate import interp1d
+
+import bluesky.plans as bp
+import bluesky.plan_stubs as bps
+from bluesky.utils import FailedStatus
+from ophyd.utils import WaitTimeoutError
+
+from start_bs import db, gov_robot
+from bluesky_env.devices.top_align import top_aligner_fast, top_aligner_slow, gonio
+from bluesky_env.devices.auto_center import work_pos
+from bluesky_env.plans.utils import mv_with_retry, mvr_with_retry
+import gov_lib
+
+def topview_optimized():
+
+    def inner_pseudo_fly_scan(*args, **kwargs):
+        scan_uid = yield from bp.count(*args, **kwargs)
+        omegas = db[scan_uid].table()[
+            top_aligner_fast.zebra.pos_capt.data.enc4.name
+        ][1]
+
+        d = np.pi / 180
+
+        # rot axis calculation, use linear regression
+        A_rot = np.matrix(
+            [[np.cos(omega * d), np.sin(omega * d), 1] for omega in omegas]
+        )
+
+        b_rot = db[scan_uid].table()[top_aligner_fast.topcam.out9_buffer.name][
+            1
+        ]
+        p = (
+            np.linalg.inv(A_rot.transpose() * A_rot)
+            * A_rot.transpose()
+            * np.matrix(b_rot).transpose()
+        )
+
+        delta_z_pix, delta_y_pix, rot_axis_pix = p[0], p[1], p[2]
+        delta_y, delta_z = (
+            delta_y_pix / top_aligner_fast.topcam.pix_per_um.get(),
+            delta_z_pix / top_aligner_fast.topcam.pix_per_um.get(),
+        )
+
+        # face on calculation
+        b = db[scan_uid].table()[top_aligner_fast.topcam.out10_buffer.name][1]
+
+        sample = 300
+        f_splines = interp1d(omegas, b)
+        b_splines = f_splines(np.linspace(omegas[0], omegas[-1], sample))
+        omega_min = np.linspace(omegas[0], omegas[-1], sample)[
+            b_splines.argmin()
+        ]
+        print(f"SPLINES / {omega_min}")
```

Comment:
log as debug?

Reply:
Added to logger.debug instead of print in 63ccbc857f8414e0fe27da5ee59e7cb547740872